pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -188,6 +181,31 @@
           }
           break;
 
+        case ':': // possibly named placerholder start
+          if (!isBatchedReWriteConfigured || !withParameters) {
+
+            nativeSql.append(aChars, fragmentStart, i - fragmentStart);
+            int end = Parser.parseNamedPlaceholder(aChars, i);
+            if (end != i) {
+              fragmentStart = i;
+
+              // Skip the colon from the captured name.
+              final String bindName = query.substring(fragmentStart + 1, end + 1);
+              final int bindIndex = paramCtx.addNamedParameter(nativeSql.length(), bindName);
+              nativeSql.append("$").append(bindIndex);
+
+              fragmentStart = end + 1;
+            } else {
+              // Skip past '::' to avoid false start on the second colon.
+              if (i + 1 < aChars.length && aChars[i + 1] == ':') {
```

Comment:
There are other tokens than `::` that are not parameter placeholders, such as e.g. `:=`.